### ISSUE TITLE
dt-options and dt-column-defs directives not recoginized

### ISSUE THREAD

**Issue by Sighan** [l-lin/angular-datatables#429]
I am trying to write a directive wrapping angular-datatables, so that I only have to create an Element of that directive and pass the data. The directive itself should take over building the table and applying the options.

The table I am creating does however not apply any options I pass to it, nor do the column-defs get applied. I create those Objects using DTOptionsBuilder and DTColumnDefBuilder, I tried to stay as close as possible to the example on https://l-lin.github.io/angular-datatables/#/angularWayWithOptions.

If you could take a look at my plunker: http://plnkr.co/edit/pu0BtIDNks1zhTT64JXN and see if you can spot what's wrong with that, I would be very grateful.

**Comment by l-lin** [comment]
Use the domain `https://rawgithub.com` instead of `https://raw.githubusercontent.com` when importing JS files because the latter will provide you a response with a Content-Type as `text/plain; charset=utf-8` whereas the first will provide you a response with a Content-Type as `application/javascript;charset=utf-8`.

**Comment by Sighan** [comment]
Hey l-lin,

thank you for your quick response. That is a good suggestion, I have removed the Github link and instead created a local file containing angular datatables in order to eliminate this as an error source.

Sadly, that did not solve the error. I have however since found a solution myself. For anyone interested:

Angular datatables only reads the dt-options parameter upon creation. Since I used the link function to generate the options, they where not defined at the time the datatable was rendered, and adding them later did not cause datatables to apply those changes (that might be something to add sometimes later, with a $scope.$watch('dtOptions') or something like that :) ).
So the solution is to use the controller function instead of the link function of the directive.
